book/entities.rst

```diff
@@ -0,0 +1,1182 @@
+        * @ORM\Table(name="acme_hotel")
+        */
+       class Hotel extend ExtendHotel
+       {
+           /**
+            * @ORM\Id
+            * @ORM\Column(type="integer")
+            * @ORM\GeneratedValue(strategy="AUTO")
+            */
+           private $id;
+
+           /**
+            * @ORM\Column(type="string", length=255)
+            */
+           private $name;
+
+           public function getId()
+           {
+               return $this->id;
+           }
+
+           public function getName()
+           {
+               return $this->name;
+           }
+
+           public function setName($name)
+           {
+               $this->name = $name;
+           }
+       }
+
+#. Add new properties using Oro migrations::
+
+       // src/Acme/DemoBundle/Migraions/Schema/v2_0;
+       namespace Acme\DemoBundle\Migrations\Schema\v2_0;
+
+       use Doctrine\DBAL\Schema\Schema;
+       use Oro\Bundle\MigrationBundle\Migration\Migration;
+       use Oro\Bundle\MigrationBundle\Migration\QueryBag;
+       use Oro\Bundle\EntityExtendBundle\EntityConfig\ExtendScope;
+
+       class HotelRankingColumn implements Migration
+       {
+           /**
+            * @inheritdoc
+            */
+           public function up(Schema $schema, QueryBag $queries)
+           {
+               $table = $schema->getTable('acme_hotel');
+               $table->addColumn(
+                   'hotel_rating',
+                   'string',
+                   array('oro_options' => array(
+                       'extend' => array(
+                           'is_extend' => true,
+                           'owner' => ExtendScope::OWNER_CUSTOM
+                       ),
+                       'entity' => array('label' => 'Hotel rating'),
+                       'datagrid' => array('is_visible' => false)
+                   ))
+               );
+           }
+       }
+
+   The example above adds a new column ``hotel_ranking``. The third parameter configures the column
+   as an extended field. Using the ``ExtendScope::OWNER_CUSTOM`` owner in the ``oro_options`` key,
+   makes the column visible and configurable in the UI. You can use ``ExtendScope::OWNER_SYSTEM``
+   instead. In that case, it cannot be removed by an administrator.
+
+   Note that this property is neither present in the ``Hotel`` entity class nor in the
+   ``ExtendHotel`` class in your bundle, but it will only be part of the ``ExtendHotel`` class that
+   will be generated in your application cache.
+
+#. Finally, load the changed configuration using the ``oro:migration:load`` command:
+
+   .. code-block:: bash
+
+       $ php app/console oro:migration:load
+
+   This command updates the database schema and generates the real implementation of the
+   ``ExtendHotel`` class in the application cache as well.
+
+.. note::
+
+    You can add, modify and remove custom fields in the UI under *System*/*Entities*/*Entity Management*.
+
+Managing Relations between Entities
```

Comment:
The associations are special type of Doctrine relations which can be used if the target side if the relation is unknown, can be changed or several entities can be used as the target. In most cases it will be better to use usual Doctrine relations, like many-to-one, many-to-many, one-to-many. The usual relations can be added by [ExtendExtension](https://github.com/orocrm/platform/blob/master/src/Oro/Bundle/EntityExtendBundle/Migration/Extension/ExtendExtension.php#L352-L591).

I guess it will be better to add two sections, one to describe usual relations, another to describe associations.
Also I guess that it will be good to describe 'option set' field type because it can be useful in many cases (it can be added by [ExtendExtension](https://github.com/orocrm/platform/blob/master/src/Oro/Bundle/EntityExtendBundle/Migration/Extension/ExtendExtension.php#L114-L324))

Please let me known if you have any questions.